grc/gui/Application.py

```diff
@@ -264,132 +264,131 @@ def flow_graph_update(fg=flow_graph):
+            for block in flow_graph.selected_blocks():
 
-                            # Check for string variables within the blocks
-                            for param in block.params.values():
-                                for variable in flow_graph.get_variables():
-                                    # If a block parameter exists that is a variable, create a parameter for it
-                                    if param.get_value() == variable.name:
-                                        params.append(param.get_value())
-                                for flow_param in flow_graph.get_parameters():
-                                    # If a block parameter exists that is a parameter, create a parameter for it
-                                    if param.get_value() == flow_param.name:
-                                        params.append(param.get_value())
+                # Check for string variables within the blocks
+                for param in block.params.values():
+                    for variable in flow_graph.get_variables():
+                        # If a block parameter exists that is a variable, create a parameter for it
+                        if param.get_value() == variable.name:
+                            params.append(param.get_value())
+                    for flow_param in flow_graph.get_parameters():
+                        # If a block parameter exists that is a parameter, create a parameter for it
+                        if param.get_value() == flow_param.name:
+                            params.append(param.get_value())
 
+                # keep track of x,y mins for pasting later
+                (x,y) = block.coordinate
+                if x < x_min:
+                    x_min = x
+                if y < y_min:
+                    y_min = y
 
-                            # keep track of x,y mins for pasting later
-                            (x,y) = block.coordinate
-                            if x < x_min:
-                                x_min = x
-                            if y < y_min:
-                                y_min = y
+                for connection in block.connections:
 
-                            for connection in block.connections:
+                    # Get id of connected blocks
+                    source_id = connection.source_block.name
+                    sink_id = connection.sink_block.name
 
-                                # Get id of connected blocks
-                                source_id = connection.source_block.name
-                                sink_id = connection.sink_block.name
+                    # If connected block is not in the list of selected blocks create a pad for it
+                    if flow_graph.get_block(source_id) not in flow_graph.selected_blocks():
+                        pads.append({'key': connection.sink_port.key, 'coord': connection.source_port.coordinate, 'block_id' : block.name, 'direction': 'source'})
```

Comment:
I'll argue that if you do a formatting commit, please make it nice :)  Line breaks at the commata, or would any python stylist/linter prefer the single line style?

not a blocker, though.